cmake/modules/AddSwift.cmake

```diff
@@ -124,9 +124,9 @@ function(_add_variant_c_compile_link_flags)
   endif()
 
   if(IS_DARWIN)
-    list(APPEND result "-isysroot" "${SWIFT_SDK_${CFLAGS_SDK}_PATH}")
-  elseif(NOT SWIFT_COMPILER_IS_MSVC_LIKE AND NOT "${SWIFT_SDK_${CFLAGS_SDK}_PATH}" STREQUAL "/")
-    list(APPEND result "--sysroot=${SWIFT_SDK_${CFLAGS_SDK}_PATH}")
+    list(APPEND result "-isysroot" "${SWIFT_SDK_${CFLAGS_SDK}_ARCH_${CFLAGS_ARCH}_PATH}")
+  elseif(NOT SWIFT_COMPILER_IS_MSVC_LIKE AND NOT "${SWIFT_SDK_${CFLAGS_SDK}_ARCH_${CFLAGS_ARCH}_PATH}" STREQUAL "/")
+    list(APPEND result "--sysroot=${SWIFT_SDK_${CFLAGS_SDK}_ARCH_${CFLAGS_ARCH}_PATH}")
```

Comment:
Might be nice to extract the value:

    set(_sysroot ${SWIFT_SDK_${CFALGS_SDK}_ARCH_${CFLAGS_ARCH}_PATH})
    if(IS_DARWIN)
      list(APPEND result -isysroot;${_sysroot})
    elseif(NOT SWIFT_COMPILER_IS_MSVC_LIKE AND NOT ${_sysroot} STREQUAL "/")
      list(APPEND result --sysroot=${_sysroot})
    endif()